WIP: Rules RBAC Role Updates

denar50 opened the pull request:
## Summary

Summarize your PR. If it involves visual changes include a screenshot or gif.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

denar50 committed:
Updates roles/tests/test roles to use new Features

This updates prebuilt roles and those used in tests to:

* Reference siemv5 instead of the older siemv4
* Add the new `rules:read` or `rules:all` feature where appropriate

Without this change, our tests are implicitly testing the
`replacedBy`/"migration" path that existing users will follow. With that
version of the code being green, we can have confidence in the behavior
for existing users, and can then update our tests to use the latest
features, here.

rylnd commented on x-pack/solutions/security/plugins/security_solution/scripts/endpoint/common/roles_users/serverless/es_serverless_resources/roles.yml:
```diff
@@ -815,24 +810,24 @@ endpoint_operations_analyst:
+        - feature_siemV5.execute_operations_all # Execute
+        - feature_siemV5.scan_operations_all
+        - feature_siemV5.workflow_insights_all
+        - feature_securitySolutionRulesV1.all
```
`endpoint_operations_analyst` should only be able to read rules, according to [this](https://docs.google.com/spreadsheets/d/16aGow187AunLCBFZLlbVyS81iQNuMpNxd96LOerWj4c/edit?gid=1936689222#gid=1936689222) spreadsheet. I've updated this on the corresponding [project-controller PR](https://github.com/elastic/elasticsearch-controller/pull/1400); pushing an update here now, too.

rylnd committed:
Endpoint Operations Analyst can only read rules.

Reference:
https://docs.google.com/spreadsheets/d/16aGow187AunLCBFZLlbVyS81iQNuMpNxd96LOerWj4c

rylnd commented:
Closing in favor of https://github.com/elastic/kibana/pull/245576.
